### DISCUSSION TITLE
cpp as makefile

### DISCUSSION THREAD

**Discussion by ghost** [github-linguist/linguist#6324] [Classification] [answered]
### Additional notes

kind of weird.

https://github.com/eladkarako/nothing_vc6

![2023-03-16_132603](https://user-images.githubusercontent.com/415238/225603442-680df4d3-d907-472d-a827-35cc0ba645b5.png)
![2023-03-16_132644](https://user-images.githubusercontent.com/415238/225603445-217d1411-594d-4d37-b099-143f1ff7b1f0.png)
![2023-03-16_132801](https://user-images.githubusercontent.com/415238/225603448-91e2a0f7-3c1c-43bd-b8a6-1770e12cafca.png)

**Comment by lildude** [comment, marked as answer]
Nothing weird at all... as you would have read in the "[How Linguist works](https://github.com/github/linguist/blob/master/docs/how-linguist-works.md)" doc:

> The percentages are calculated based on the bytes of code for each language as reported by the [List Languages](https://docs.github.com/rest/reference/repos#list-repository-languages) API.

The Makefile is the largest file by bytes of code in the entire repo of those files that Linguist knows about:

```console
➜ github-linguist nothing_vc6/ --breakdown
82.34%  5163       Makefile
11.91%  747        C
5.74%   360        C++

C++:
source/StdAfx.cpp
source/nothing_vc6.cpp

C:
source/StdAfx.h

Makefile:
source/nothing_vc6.mak

➜ /bin/ls -l nothing_vc6/source/*.(mak|cpp|h)
-rw-r--r--  1 lildude  staff   290 16 Mar 11:32 nothing_vc6/source/StdAfx.cpp
-rw-r--r--  1 lildude  staff   747 16 Mar 11:32 nothing_vc6/source/StdAfx.h
-rw-r--r--  1 lildude  staff    70 16 Mar 11:32 nothing_vc6/source/nothing_vc6.cpp
-rw-r--r--  1 lildude  staff  5163 16 Mar 11:32 nothing_vc6/source/nothing_vc6.mak

➜
```
